File: L1. Contains Duplicate/ContainsDuplicate.java
// https://leetcode.com/problems/contains-duplicate/
// LC217: Contains Duplicate
// NC Category: Arrays & Hashing.
// Time/space: Quadratic time, constant extra-space.
// Solution kind: Baseline.
// Approach: Brute force.
class ContainsDuplicate {
    public boolean containsDuplicate(int[] nums) {
        int n = nums.length;
        for (int i=0; i<n; i++) {
            for (int j=i+1; j<n; j++) {
                if (nums[i] == nums[j]) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: contains-duplicate-linearish-time/ContainsDuplicate.cpp
// https://leetcode.com/problems/contains-duplicate
// Linearish time, linearish extra-space.
// Concepts: Hashing, Symmetry.
class ContainsDuplicate {
public:
    bool containsDuplicate(vector<int>& nums) {
        std::unordered_set<int> set;
        for (int num : nums) {
            if (set.contains(num)) {
                return true;
            }
            set.insert(num);
        }
        return false;
    }
};
